contracts/governance/extensions/GovernorVotesSuperQuorumFraction.sol

```diff
@@ -0,0 +1,131 @@
+        return _superQuorumNumeratorHistory.latest();
+    }
+
+    /**
+     * @dev Returns the super quorum numerator at a specific timepoint.
+     */
+    function superQuorumNumerator(uint256 timepoint) public view virtual returns (uint256) {
+        return _optimisticUpperLookupRecent(_superQuorumNumeratorHistory, timepoint);
+    }
+
+    /**
+     * @dev Returns the super quorum for a timepoint, in terms of number of votes: `supply * numerator / denominator`.
+     */
+    function superQuorum(uint256 timepoint) public view virtual override returns (uint256) {
+        return Math.mulDiv(token().getPastTotalSupply(timepoint), superQuorumNumerator(timepoint), quorumDenominator());
+    }
+
+    /**
+     * @dev Changes the super quorum numerator.
+     *
+     * Emits a {SuperQuorumNumeratorUpdated} event.
+     *
+     * Requirements:
+     *
+     * - Must be called through a governance proposal.
+     * - New super quorum numerator must be smaller or equal to the denominator.
+     * - New super quorum numerator must be bigger than the quorum numerator.
+     */
+    function updateSuperQuorumNumerator(uint256 newSuperQuorumNumerator) external virtual onlyGovernance {
+        _updateSuperQuorumNumerator(newSuperQuorumNumerator);
+    }
+
+    /**
+     * @dev Changes the super quorum numerator.
+     *
+     * Emits a {SuperQuorumNumeratorUpdated} event.
+     *
+     * Requirements:
+     *
+     * - New super quorum numerator must be smaller or equal to the denominator.
+     * - New super quorum numerator must be bigger than the quorum numerator.
+     */
+    function _updateSuperQuorumNumerator(uint256 newSuperQuorumNumerator) internal virtual {
+        uint256 denominator = quorumDenominator();
+        if (newSuperQuorumNumerator > denominator) {
+            revert GovernorInvalidSuperQuorumFraction(newSuperQuorumNumerator, denominator);
+        }
+        uint256 quorumNumerator = quorumNumerator();
+        if (newSuperQuorumNumerator <= quorumNumerator) {
+            revert GovernorInvalidSuperQuorumTooSmall(newSuperQuorumNumerator, quorumNumerator);
+        }
+
+        uint256 oldSuperQuorumNumerator = _superQuorumNumeratorHistory.latest();
+        _superQuorumNumeratorHistory.push(clock(), SafeCast.toUint208(newSuperQuorumNumerator));
+
+        emit SuperQuorumNumeratorUpdated(oldSuperQuorumNumerator, newSuperQuorumNumerator);
+    }
+
+    /**
+     * @dev Overrides {GovernorVotesQuorumFraction._updateQuorumNumerator} to ensure the super
+     * quorum numerator is bigger than the quorum numerator.
+     */
+    function _updateQuorumNumerator(uint256 newQuorumNumerator) internal virtual override {
+        uint256 superQuorumNumerator_ = superQuorumNumerator();
+        // Ignoring a super quorum of 0 as it is the initial value when the contract is deployed.
+        if (superQuorumNumerator_ != 0 && newQuorumNumerator >= superQuorumNumerator_) {
```

Comment:
The initial value is set in the constructor and is never ignored if it is 0 other than here. Given that there are existing constraints on the super quorum being greater than the quorum (even from the time of construction), we can eliminate this check.

```suggestion
        uint256 superQuorumNumerator_ = superQuorumNumerator();
        if (newQuorumNumerator >= superQuorumNumerator_) {
```

Reply:
This will not work, see related discussion above: https://github.com/OpenZeppelin/openzeppelin-contracts/pull/5492#discussion_r1949166979

Reply:
Ah yes sorry for repeating this issue.